src/hotspot/share/opto/doCall.cpp

```diff
@@ -790,46 +790,47 @@ void Parse::catch_call_exceptions(ciExceptionHandlerStream& handlers) {
   Node* i_o = this->i_o();
 
   // Add a CatchNode.
-  GrowableArray<int>* bcis = new (C->node_arena()) GrowableArray<int>(C->node_arena(), 8, 0, -1);
-  GrowableArray<const Type*>* extypes = new (C->node_arena()) GrowableArray<const Type*>(C->node_arena(), 8, 0, nullptr);
-  GrowableArray<int>* saw_unloaded = new (C->node_arena()) GrowableArray<int>(C->node_arena(), 8, 0, 0);
+  Arena tmp_mem{mtCompiler};
+  GrowableArray<int> bcis(&tmp_mem, 8, 0, -1);
+  GrowableArray<const Type*> extypes(&tmp_mem, 8, 0, nullptr);
+  GrowableArray<int> saw_unloaded(&tmp_mem, 8, 0, -1);
 
   bool default_handler = false;
   for (; !handlers.is_done(); handlers.next()) {
-    ciExceptionHandler* h        = handlers.handler();
-    int                 h_bci    = h->handler_bci();
-    ciInstanceKlass*    h_klass  = h->is_catch_all() ? env()->Throwable_klass() : h->catch_klass();
+    ciExceptionHandler* h       = handlers.handler();
+    int                 h_bci   = h->handler_bci();
+    ciInstanceKlass*    h_klass = h->is_catch_all() ? env()->Throwable_klass() : h->catch_klass();
     // Do not introduce unloaded exception types into the graph:
     if (!h_klass->is_loaded()) {
-      if (saw_unloaded->contains(h_bci)) {
+      if (saw_unloaded.contains(h_bci)) {
         /* We've already seen an unloaded exception with h_bci,
            so don't duplicate. Duplication will cause the CatchNode to be
            unnecessarily large. See 4713716. */
         continue;
       } else {
-        saw_unloaded->append(h_bci);
+        saw_unloaded.append(h_bci);
       }
     }
-    const Type*         h_extype = TypeOopPtr::make_from_klass(h_klass);
+    const Type* h_extype = TypeOopPtr::make_from_klass(h_klass);
```

Comment:
Removed the alignment here, as it's far away from the previously declared variables in this scope.